OpenRA.Game/Support/HttpQueryBuilder.cs

```diff
@@ -0,0 +1,61 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenRA.Support
+{
+	public class HttpQueryBuilder : IEnumerable
+	{
+		readonly string url;
+		readonly List<Parameter> parameters = new List<Parameter>();
+
+		public HttpQueryBuilder(string url)
+		{
+			this.url = url;
+		}
+
+		public void Add(string name, object value)
+		{
+			parameters.Add(new Parameter
+			{
+				Name = name,
+				Value = Uri.EscapeUriString(value.ToString())
+			});
+		}
+
+		public override string ToString()
+		{
+			var builder = new StringBuilder(url);
+
+			builder.Append("?");
+
+			foreach (var parameter in parameters)
+				builder.Append($"{parameter.Name}={parameter.Value}&");
```

Comment:
Won't this leave a stray `&` at the end of the query string?
Do we care?

Reply:
FWIW it doesn't seem to hurt 🤷